pkg/controller/attach_detach.go

```diff
@@ -117,8 +117,15 @@ func (r *ReconcileAttachDetach) Reconcile(ctx context.Context, request reconcile
 		}
 		// promotion request
 	} else if azVolumeAttachment.Spec.RequestedRole != azVolumeAttachment.Status.Detail.Role {
-		if err := r.promote(ctx, azVolumeAttachment); err != nil {
-			return reconcileReturnOnError(azVolumeAttachment, "promote", err, r.retryInfo)
+		switch azVolumeAttachment.Spec.RequestedRole {
+		case v1alpha1.PrimaryRole:
+			if err := r.promote(ctx, azVolumeAttachment); err != nil {
+				return reconcileReturnOnError(azVolumeAttachment, "promote", err, r.retryInfo)
+			}
+		case v1alpha1.ReplicaRole:
+			if err := r.demote(ctx, azVolumeAttachment); err != nil {
```

Comment:
We queue replica garbage collection in case no new primary is selected? I don't believe we can only rely of the volume being deleted to trigger replica detach because there is no guarantee the PVC/PV will be deleted after the workload stops.

Reply:
In this circumstance, I think it makes sense that we queue garbage collection. This is no different from our usual scenario, where we queue garbage collection upon deletion of primary AzVolumeAttachment. We give 5 minute grace period before we initiate garbage collection, and if this volume ends up not being used for any workload within that grace period, the driver conclusively decides that it would be okay to get rid of all replica AzVolumeAttachment.